research/python/src/data/provider.py

```
@@ -0,0 +1,1077 @@
+        return None
+
+    def _retry_operation(self, operation, *args, **kwargs):
+        """Retry an operation with exponential backoff."""
+        import time
+
+        for attempt in range(MAX_RETRY_ATTEMPTS):
+            try:
+                return operation(*args, **kwargs)
+            except Exception as e:
+                if attempt == MAX_RETRY_ATTEMPTS - 1:
+                    raise e
```

Ruff (0.14.10)

Use `raise` without specifying exception name

Remove exception name

(TRY201)